test/async/retry_test.dart

```diff
@@ -46,5 +48,12 @@ void main() {
               timeout: const Duration(milliseconds: 1)),
           throwsA(equals('error')));
     });
+
+    test('should timeout long tasks', () async {
+      var task = () => new Completer().future;
+
+      expect(retry(task, timeout: const Duration(milliseconds: 100)),
```

Comment:
I tried rewriting the tests using FakeAsync, and I don't think it buys us much. The crucial pieces are calls to Future#timeout and those don't take a Clock so we can't force them to trigger or to wait.